src/v2/providers/dataconnect.ts

```diff
@@ -0,0 +1,365 @@
+  /** The location of the Firebase Data Connect instance */
+  location: string;
+  /** The project identifier */
+  project: string;
+  /**
+   * An object containing the values of the path patterns.
+   * Only named capture groups will be populated - {key}, {key=*}, {key=**}.
+   */
+  params: Params;
+  /** The type of principal that triggered the event */
+  authType: AuthType;
+  /** The unique identifier for the principal */
+  authId?: string;
+}
+
+/**
+ * Event handler that triggers when a mutation is executed in Firebase Data Connect.
+ *
+ * @param mutation - The mutation path to trigger on.
+ * @param handler - Event handler which is run every time a mutation is executed.
+ */
+export function onMutationExecuted<
+  Mutation extends string,
+  Variables = unknown,
+  ResponseData = unknown
+>(
+  mutation: Mutation,
+  handler: (
+    event: DataConnectEvent<MutationEventData<Variables, ResponseData>, DataConnectParams<Mutation>>
+  ) => unknown | Promise<unknown>
+): CloudFunction<
+  DataConnectEvent<MutationEventData<Variables, ResponseData>, DataConnectParams<Mutation>>
+>;
+
+/**
+ * Event handler that triggers when a mutation is executed in Firebase Data Connect.
+ *
+ * @param opts - Options that can be set on an individual event-handling function.
+ * @param handler - Event handler which is run every time a mutation is executed.
+ */
+export function onMutationExecuted<
+  Options extends OperationOptions,
+  Variables = unknown,
+  ResponseData = unknown
+>(
+  opts: Options,
+  handler: (
+    event: DataConnectEvent<MutationEventData<Variables, ResponseData>, DataConnectParams<Options>>
+  ) => unknown | Promise<unknown>
+): CloudFunction<
+  DataConnectEvent<MutationEventData<Variables, ResponseData>, DataConnectParams<Options>>
+>;
+
+/**
+ * Event handler that triggers when a mutation is executed in Firebase Data Connect.
+ *
+ * @param mutationOrOpts - Options or string mutation path.
+ * @param handler - Event handler which is run every time a mutation is executed.
+ */
+export function onMutationExecuted<
+  PathPatternOrOptions extends string | OperationOptions<string, string, string>,
+  Variables = unknown,
+  ResponseData = unknown
+>(
+  mutationOrOpts: PathPatternOrOptions,
+  handler: (
+    event: DataConnectEvent<
+      MutationEventData<Variables, ResponseData>,
+      DataConnectParams<PathPatternOrOptions>
+    >
+  ) => unknown | Promise<unknown>
+): CloudFunction<
+  DataConnectEvent<
+    MutationEventData<Variables, ResponseData>,
+    DataConnectParams<PathPatternOrOptions>
+  >
+> {
+  return onOperation<Variables, ResponseData, PathPatternOrOptions>(
+    mutationExecutedEventType,
+    mutationOrOpts,
+    handler
+  );
+}
+
+function getOpts(mutationOrOpts: string | OperationOptions) {
+  const operationRegex = new RegExp(
+    "locations/([^/]+)/services/([^/]+)/connectors/([^/]*)/operations/([^/]+)"
```

Comment:
@taeold - Could use some feedback on recent commit changes. With the recent changes, a customer can set the region by doing one of:

1. Passing in region to GlobalOptions
2. Passing in region to OperationOptions, e.g.

```
onMutationExecuted({
  region: "us-west4"
}, (event) => {});
```

3. Pass in region to mutation "resource"

```
onMutationExecuted(
  "locations/us-west4/services/*/connectors/*/operations/*",
  (event) => {}
);
```

Not sure about option 3, and it strays a bit from the original API proposal. Other products seem to only expose options 1 and 2 as ways to set the region. WDYT? 

CC: @yuchenshi  - if you have thoughts too from today's bug bash

Reply:
Discussed offline w/ @taeold capturing some of the points here:

No contention for 1) and 2) since that's what exists in SDK today. As for 3) similar concerns around inconsistency with other triggers.

Let's refrain from introducing the "locations/..." for now. I'll include a callout in our docs that the default is us-central1 and can adjust according to customer feedback. I'll close this thread once I've made the appropriate updates